### PR TITLE
Nested result

### PR DESCRIPTION
**Pull request by ajanata**
To make reporting results to Slack easier, switch to a proper result hierarchy instead of just nested maps. The top level now has a success indicator as well as execution duration. Each package does as well, and each test can have subtests nested inside subtests to an unlimited depth. Also fixes a bug when getting package names.

The ordered map is now only needed to know which tests to run, and not for the test results, so moving it to internal to discourage its use elsewhere.

### CODE REVIEW

File: register.go [hunk 1]
```diff
@@ -18,17 +18,21 @@ func getCallerPackage() string {
...
 	return pkg
 }
 
+func packageAndFuncNameToPackage(full string) string {
+	// skip to after the package.
+	i := strings.LastIndex(full, "/")
+	// remove the function name (which is almost certainly "init") and leave just the package name.
+	// as an example, this function is `github.com/gametimesf/testy.RegisterTest`.
+	// we cannot just do LastIndex for . because anonymous functions will end up with multiple . in their name
+	j := strings.Index(full[i:], ".")
```
Comment (ajanata) [inline register.go:32]: this was previously `:i` which was the bug (it was finding the `.` after `github` and using that as the limit for package names, which truncated longer names)

File: run.go [hunk 2]
```diff
@@ -170,31 +180,30 @@ func Run() orderedmap.OrderedMap[string, orderedmap.OrderedMap[string, TestResul
...
-							Passed:  false,
-							Dur:     old.Dur,
-							Msgs: append(old.Msgs, append(testHelperT.msgs, Msg{
-								Msg:   fmt.Sprintf("%v", afterTestErr),
-								Level: LevelError,
-							})...),
-						}
+						pkgAnyFailures = true
+						// update test results marking it failed and with this panic message.
+						r := &pkgResults.Subtests[len(pkgResults.Subtests)-1]
```
Comment (waitstime) [inline run.go:185]: This pattern is fine but feels awkward. Maybe a function on a typed []TestResult receiver to access last element?

**Review by waitstime — APPROVED** [review]
Looks good just one nit

File: run.go (hunk 2 above)
Comment (ajanata) [reply run.go:185]: I'm not sold on making an opaque type just to avoid slice length calculations in 2 locations

**Review by kendallm — APPROVED** [review]

**Pull request merged**